### PR TITLE
docs tests CI

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
#### Summary of Changes (What does this PR do?)
- Adds tests to the docs CI pipeline

#### Status of maturity (all need to be checked before merging):
- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [x] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?
- Checkout the the GitHub run workflows
- You should see tests of the docs folder tested too

#### What are the relevant tickets?
- n/a

#### Screenshots (optional)

### CODE REVIEW

File: docs/cypress/support/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      return;
+    }
+    return original.apply(this, arguments);
+  };
+
+  Object.defineProperty(window, "onbeforeunload", {
+    get: function () {},
+    set: function () {},
+  });
+});
```
Comment (123MwanjeMike) [inline docs/cypress/support/index.js:15]: Function to allow tests to run on electron and firefox

File: docs/package.json [hunk 2]
```diff
@@ -16,7 +16,9 @@
   "scripts": {
     "docker:build": "docker build -f Dockerfile -t docsify-test:local .",
     "docker:clean": "docker rmi docsify-test:local",
-    "docker:cli": "docker run --rm -it -p 3000:3000 docsify-test:local"
+    "docker:cli": "docker run --rm -it -p 3000:3000 docsify-test:local",
+    "preview": "docsify serve .",
+    "test": "cypress open"
```
Comment (123MwanjeMike) [inline docs/package.json:21]: Add preview and test commands

File: docs/index.html [hunk 3]
```diff
@@ -28,7 +28,6 @@
         loadNavbar: true,
         loadSidebar: true,
         subMaxLevel: 3,
-        search: ["/", "api/*", "mobile_app/*", "platform/*"],
```
Comment (123MwanjeMike) [inline docs/index.html:31]: No longer needed with the recent version update

File: docs/cypress.config.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 const { defineConfig } = require("cypress");
 
 module.exports = defineConfig({
+  projectId: "v2fvjt",
```
Comment (123MwanjeMike) [inline docs/cypress.config.js:4]: app preject ID on cypress

File: .github/workflows/test-changes.yml [hunk 5]
```diff
@@ -70,11 +75,40 @@ jobs:
...
+        uses: cypress-io/github-action@v4
+        with:
+          working-directory: ./docs/
+          start: yarn preview
+          wait-on: http://localhost:3000
+          browser: chrome
+          record: true
+        env:
+          CYPRESS_RECORD_KEY: ${{ secrets.DOCS_CYPRESS_RECORD_KEY }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (123MwanjeMike) [inline .github/workflows/test-changes.yml:108]: Test docs changes

File: .github/workflows/test-changes.yml [hunk 6]
```diff
@@ -70,11 +75,40 @@ jobs:
           browser: chrome
           record: true
         env:
-          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+          CYPRESS_RECORD_KEY: ${{ secrets.ANALYTICS_PLATFORM_CYPRESS_RECORD_KEY }}
```
Comment (123MwanjeMike) [inline .github/workflows/test-changes.yml:78]: Use project specific record key name

**Review by 123MwanjeMike — COMMENTED** [review]
Guiding comments

**Review by Baalmart — APPROVED** [review]
This LGTM, thanks @123MwanjeMike

**Pull request merged**